### PR TITLE
perf(attention): share QSA score tiles across query rows

### PR DESCRIPTION
**Pull request by tuanzhangCS**
## Summary

Reduce Qwen4-Exp QSA logits-scoring cost for long single-request prefill while keeping one scoring path for eager, captured, grouped, and ragged layouts.

- Pass the existing uniform query width into the kernel. Small groups share K across up to four queries; long NVIDIA runs with four 128-dimensional index heads share each K tile across 32 queries. The selected width is always a divisor of the request width, so a group cannot cross request boundaries.
- Bound an extend batch's compressed block-table view from host-known prefix-plus-query lengths. This changes only the current metadata view; the persistent allocation, page mapping, and decode/CUDA-graph capacity remain unchanged.
- Skip Q/K loads and the dot product for score tiles beyond every query's complete-block frontier, while still writing `-inf`.
- Retain the existing one-row path for ragged or mixed layouts and preserve each row's frontier and top-k selection. The streaming top-k path is unchanged.

Grouping may change FP32 dot/reduction rounding. Correctness is checked against the FP32 reference with `rtol=1e-5, atol=1e-4`; selection is checked against each layout's own scores and tie-breaking.

## Performance

Measured on NVIDIA B300 with PyTorch 2.14.0+cu130, CUDA 13.0, and tokenspeed-triton 3.8.10.post20260920. The target is one 16,384-token request, BF16, four 128-dimensional index heads, strided Q, page size 64, 4,096 compressed score columns, and causal per-row frontiers. PDL is enabled.

The benchmark measures only `_qwen4_exp_qsa_score_blocks_kernel`, after warmup; allocation, compilation, and persistent-radix top-k are excluded. Median of 21 CUDA-event samples:

| Query group | Grid | Score latency | Registers | Shared memory |
| ---: | ---: | ---: | ---: | ---: |
| 4 (previous PR path) | `(4096, 32)` | 445.9 µs | 80 | 36,864 B |
| 32 | `(512, 32)` | 205.8 µs | 168 | 65,544 B |

The long-run specialization is 2.17x faster (53.8% lower score latency), with no spills. These results cover the listed kernel shape rather than end-to-end inference throughput.

## Test Plan

Validated on NVIDIA B300, Python 3.12, and PyTorch 2.14.0+cu130:

- `pre-commit run --all-files` — passed.
- `CUDA_VISIBLE_DEVICES=5 PYTHONPATH=python:tokenspeed-kernel/python pytest -q tokenspeed-kernel/test/ops/test_qwen4_exp_qsa.py` — 88 passed, 4 skipped.
- `CUDA_VISIBLE_DEVICES=5 PYTHONPATH=python:tokenspeed-kernel/python pytest -q tokenspeed-kernel/test/nvidia/ops/test_qsa_pdl.py` — 30 passed.
- `CUDA_VISIBLE_DEVICES=5 PYTHONPATH=python:tokenspeed-kernel/python pytest -q test/runtime/test_qsa_backend.py test/runtime/test_qwen4_exp.py` — 157 passed.

The skipped tests require the optional native persistent-radix top-k extension, which is unavailable in this worktree. No failures.

### CODE REVIEW

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/attention/qsa/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# QSA block scoring
```
[the remark this answers is not in the document]
Comment (tuanzhangCS) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/attention/qsa/README.md]: ok

**Comment by tuanzhangCS** [comment]
> > The local environment uses PyTorch 2.13.0; the PyTorch 2.14 dependency update on main was not exercised locally.
> 
> Let's also verify with torch 2.14

ok
